marlowe-dashboard-client/src/Pickup/View.purs

```diff
@@ -0,0 +1,188 @@
+module Pickup.View (renderPickupState) where
+
+import Prelude hiding (div)
+import Css (classNames, hideWhen, toggleWhen)
+import Data.Lens (view)
+import Data.Maybe (Maybe(..), isJust, isNothing)
+import Data.Tuple (fst, snd)
+import Halogen.HTML (HTML, a, br_, button, div, footer, h2_, header, input, label, main, p, text)
+import Halogen.HTML.Events (onClick, onValueInput)
+import Halogen.HTML.Properties (InputType(..), disabled, for, href, id_, list, placeholder, readOnly, type_, value)
+import MainFrame.Lenses (_card, _screen)
+import Material.Icons as Icon
+import Pickup.Types (Action(..), Card(..), Screen(..), State)
+import WalletData.Lenses (_key, _nickname)
+import WalletData.Types (WalletLibrary, WalletNicknameKey)
+import WalletData.Validation (nicknameError)
+import WalletData.View (nicknamesDataList)
+
+renderPickupState :: forall p. WalletLibrary -> WalletNicknameKey -> State -> HTML p Action
+renderPickupState wallets newWalletNicknameKey pickupState =
+  let
+    screen = view _screen pickupState
+
+    card = view _card pickupState
```

Comment:
with these ones, I think it's better to pattern match in the function definition
```purescript
renderPickupState wallets newWalletNicknameKey { card, screen } =
```

Reply:
Maybe. I had the idea that I would just use lenses everywhere, so that the decision how to do it never got in my way, and so that if we ever change things (e.g. to a newtype) we don't have to rewrite stuff like this. 🤷

Reply:
true